cognee/api/health.py

```diff
@@ -0,0 +1,327 @@
+                status=HealthStatus.HEALTHY,
+                provider=config.graph_database_provider,
+                response_time_ms=response_time,
+                details="Schema validated",
+            )
+        except Exception as e:
+            response_time = int((time.time() - start_time) * 1000)
+            return ComponentHealth(
+                status=HealthStatus.UNHEALTHY,
+                provider="unknown",
+                response_time_ms=response_time,
+                details=f"Connection failed: {str(e)}",
+            )
+
+    async def check_file_storage(self) -> ComponentHealth:
+        """Check file storage health."""
+        start_time = time.time()
+        try:
+            import os
+            from cognee.infrastructure.files.storage.get_file_storage import get_file_storage
+            from cognee.base_config import get_base_config
+
+            base_config = get_base_config()
+            storage = get_file_storage(base_config.data_root_directory)
+
+            # Determine provider
+            provider = "s3" if base_config.data_root_directory.startswith("s3://") else "local"
+
+            # Test storage accessibility - for local storage, just check directory exists
+            if provider == "local":
+                os.makedirs(base_config.data_root_directory, exist_ok=True)
+                # Simple write/read test
+                test_file = os.path.join(base_config.data_root_directory, "health_check_test")
+                with open(test_file, "w") as f:
+                    f.write("test")
+                os.remove(test_file)
+            else:
+                # For S3, test basic operations
+                test_path = "health_check_test"
+                await storage.store(test_path, b"test")
+                await storage.delete(test_path)
+
+            response_time = int((time.time() - start_time) * 1000)
+            return ComponentHealth(
+                status=HealthStatus.HEALTHY,
+                provider=provider,
+                response_time_ms=response_time,
+                details="Storage accessible",
+            )
+        except Exception as e:
+            response_time = int((time.time() - start_time) * 1000)
+            return ComponentHealth(
+                status=HealthStatus.UNHEALTHY,
+                provider="unknown",
+                response_time_ms=response_time,
+                details=f"Storage test failed: {str(e)}",
+            )
+
+    async def check_llm_provider(self) -> ComponentHealth:
+        """Check LLM provider health (non-critical)."""
+        start_time = time.time()
+        try:
+            from cognee.infrastructure.llm.get_llm_client import get_llm_client
+            from cognee.infrastructure.llm.config import get_llm_config
+
+            config = get_llm_config()
+
+            # Test actual API connection with minimal request
+            client = get_llm_client()
+            await client.acomplete("test", max_tokens=1)
```

Comment:
this method doesn't exist, please check LLMProvider interface.

For now, you can replace this with something like `client.show_prompt("test", "test")`
```suggestion
            client.show_prompt("test", "test")
```

Reply:
(note to self) We can think about improving this by making an actual query to the LLM (+ add some caching to avoid spending quota on health check), but this check is already good as it will let us catch issues like missing keys.

<img width="1508" height="304" alt="Screenshot 2025-08-12 at 15 42 29" src="https://github.com/user-attachments/assets/a5fc6160-4425-46e8-afb2-eab0a0ade3b6" />